include/zephyr/net/tls_credentials.h

```diff
@@ -124,6 +124,37 @@ int tls_credential_get(sec_tag_t tag, enum tls_credential_type type,
  */
 int tls_credential_delete(sec_tag_t tag, enum tls_credential_type type);
 
+/**
+ * @brief Create client private key and certificate signing request (CSR).
```

Comment:
since this function does 2 things, any consideration for:

change to `tls_credential_generate_key` - with optional CSR or public key output
and/or 
allow `tls_credential_csr` to use existing credentials

Reply:
The thought has definitely crossed my mind.

My initial thought hat been to add only `tls_credentials_csr` and then later add a `tls_credential_keygen` function which only performs keygen.

But you make a very good point; Users may want `tls_credentials_csr` to support using a pre-existing private key, so long as the credentials backend can handle it.

It will be some extra work, but I'll go ahead and implement `tls_credential_keygen`, and give `tls_credentials_csr` an option to use an existing PK sectag rather than generate one. I think you are correct that this would be cleaner, and it will save us headache down the line if we implement it now.